### ISSUE TITLE
Remove all notion of LTI Params from delivery

### ISSUE THREAD

**Issue by eliknebel** [Simon-Initiative/oli-torus#1980]
In order to facilitate user masquerading and better preview functionality, all notions of lti params and accessing them should be removed from delivery. Any data required for delivery which may come from LTI Params should be modeled directly in torus and simply updated during the LTI launch.

Torus intends to support admin capability to act on behalf of, or "masquerade" as a user. Some of the torus delivery implementation relies on the LTI launch params to deliver course content. This is problematic because it requires that a user launch from their LMS in order to set these params. Furthermore, in order to access these params, the lti_1p3 library caches them and returns a key derived from unique information about a particular launch, however this key cannot be easily generated to retrieve LTI params outside the context of a launch making it difficult to masquerade as a user without a proper launch.

LTI params should not be accessed from the delivery context after a launch is complete since the data in those params might only relate to that particular launch context or time. Instead, when a LTI launch is performed, any relevant data should be translated to the appropriate torus counterpart and stored as part of the overall torus data model. The relevant data should be agnostic to how a user accessed the platform. For example, if a user launches via LTI and has a particular context role according to the LTI params, that context role should be modeled in the torus database accordingly so that an LTI launch does not need to be performed in order for the context roles of a user to be read.

In summary, any LTI params given from an LTI launch should be translated into the relevant torus data models and updated. LTI params should be decoupled from delivery. However, for informational and debugging purposes, LTI params should still be cached and associated with a particular user.

**Comment by eliknebel** [comment]
Verified, the only routes that currently require LTI params are DeliveryController :index, :select_project, :research_consent, and :create_section. These routes are LTI specific and delivery content independent and do in-fact utilize the lti params.

The remaining work here is to change how LTI params get cached by lti_1p3. Instead of lti_1p3 caching these params then returning a key, it should simply return the params and it should be up to the application (torus) to handle the caching/storage of params. A torus specific schema can then easily attach a user_id fk to the lti params record so that it can be easily associated with a specific user

**Comment by LEDThereBeLight** [comment]
Closing as resolved.
